### PR TITLE
Remove CommonExtensions.php from config to app.

### PR DESCRIPTION
**Pull request by ghost**
Sister Wikia/config PR: https://github.com/Wikia/config/pull/2389

The checks are failing because of circular dependency of app@remove-common-extensions and config@remove-common-extensions.  Tests pass on `dev` when both branches are checked out.

### CODE REVIEW

File: includes/wikia/Extensions.php [hunk 1]
```diff
@@ -0,0 +1,1892 @@
...
+/**
+ * define extension directory based on IP variable
+ */
+$wgExtensionsDirectory = $IP . "/extensions";
+
+/**
+ * $wgDefaultSkin is first set to 'oasis' in CommonSettings.php. Then the value is assigned to
+ * $wgDefaultSkinBeforeWF and $wgDefaultSkin is nullified. WikiFactor is supposed to assign
+ * some value to $wgDefaultSkin - a per-wiki setting. If it fails and at this point $wgDefaultSkin
+ * is still null, we use the cached value from $wgDefaultSkinBeforeWF.
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:29]: could we simplify this?

File: includes/wikia/Extensions.php [hunk 2]
```diff
@@ -0,0 +1,1892 @@
...
+ */
+if ( is_null( $wgDefaultSkin ) ) {
+	$wgDefaultSkin = $wgDefaultSkinBeforeWF;
+}
+
+//can access ->cat_id or ->cat_name
+$wgHub = WikiFactory::getCategory($wgCityId);
+
+# Remove any skin(s) listed in UnSkipSkins from SkipSkins
+if( !empty($wgUnSkipSkins) && is_array($wgUnSkipSkins) ) {
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:39]: this is set on 3 wikis - imho let's set `wgSkipSkins` value for those and get rid of this

File: includes/wikia/Extensions.php [hunk 3]
```diff
@@ -0,0 +1,1892 @@
...
+	$wgSkipSkins = array_diff($wgSkipSkins, $wgUnSkipSkins);
+}
+
+# Language specific settings. Currently only used for $wgExtraNamespaces.
+# Consider using /languages/messages/Wikia/ for customization.
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:44]: ☝️

File: includes/wikia/Extensions.php [hunk 4]
```diff
@@ -0,0 +1,1892 @@
...
+		$wgNamespaceAliases["Forum"] = 110;
+		$wgNamespaceAliases["Forum_talk"] = 111;
+
+		break;
+
+}
+
+// new style of per-lang config
+if (!in_array($wgLanguageCode, $wgAvailableAnswersLang) && !in_array(preg_replace("/-.*$/", "", $wgLanguageCode), $wgAvailableAnswersLang)) {
+   $wgEnableAnswersMonacoWidget = false;
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:114]: 🇲🇨 ? let's check if this still controls anything

File: includes/wikia/Extensions.php [hunk 5]
```diff
@@ -0,0 +1,1892 @@
...
+	require_once "$LW/Templates.php";
+	require_once "$LW/Hook_PreventBlanking.php";
+	require_once "$LW/Special_ArtistRedirects.php";
+	require_once "$LW/lw_spiderableBadArtists.php";
+	require_once "$LW/Special_Soapfailures.php";
+	require_once "$LW/Special_MobileSearches.php";
+	require_once "$LW/lw_impliedRedirects.php";
+	# Turn off subpages on the main namespace (otherwise every AC/DC song links back to "AC"), etc.
+	$wgNamespacesWithSubpages[ NS_MAIN ] = false;
+	require_once "$LW/LyricFind/LyricFind.setup.php";
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:160]: consider migrating these to classmap autoloader

File: includes/wikia/Extensions.php [hunk 6]
```diff
@@ -0,0 +1,1892 @@
...
+
+
+if ( !empty( $wgEnableEditPageLayoutExt ) ) {
+	include "$IP/extensions/wikia/EditPageLayout/EditPageLayout_setup.php";
+}
+
+/**
+ * load extensions by using configuration variables
+ */
+#--- 1. Special::ProtectSite
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:227]: these numbers will never ever be up to date

File: includes/wikia/Extensions.php [hunk 7]
```diff
@@ -0,0 +1,1892 @@
...
+
+#--- 28. Wikia Shared talk
+if (!empty( $wgWikiaEnableSharedTalkExt )) {
+	include( "$IP/extensions/wikia/WikiaNewtalk/wikia_newtalk.php" );
+}
+
+/**
+ * DynamicPageList & DPLForum
+ */
+if( defined( 'REBUILD_LOCALISATION_CACHE_IN_PROGRESS' ) || !empty($wgWikiaEnableDPLExt) && empty( $wgWikiaDisableAllDPLExt ) ) {
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:299]: not needed anywhere anymore due to refactor of localisation cache rebuild

File: includes/wikia/Extensions.php [hunk 8]
```diff
@@ -0,0 +1,1892 @@
...
+}
+
+include("$IP/extensions/wikia/UserProfilePageV3/UserProfilePage.setup.php");
+
+// Michał Roszka (Mix) <michal@wikia-inc.com>
+// BugId:10474
+// I want the code of the extension to be included regardless of the $wgEnableAchievementsExt.
+// The effective execution still depends on $wgEnableAchievementsExt's value.
+// Also: grep extensions/wikia/AchievementsII/Ach_setup.php for BugId:10474
+include("$IP/extensions/wikia/AchievementsII/Ach_setup.php");
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:1022]: Let's check why we need Achievements code when the extension is disabled

File: includes/wikia/Extensions.php [hunk 9]
```diff
@@ -0,0 +1,1892 @@
...
+# SponsorshipDashboard
+if (!empty($wgEnableSponsorshipDashboardExt)) {
+	include("$IP/extensions/wikia/SponsorshipDashboard/SponsorshipDashboard_setup.php" );
+	include("$IP/extensions/wikia/GoogleAnalyticsSampling/GoogleAnalyticsSampling.setup.php");
+}
+
+$wgEnableOpenXSPC = true;
+$wgEnableSpotlightsV2_GlobalNav = true;
+$wgEnableSpotlightsV2_Footer = true;
+$wgEnableSpotlightsV2_Rail = true;
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:1136]: are these still needed if they are always `true`?

File: includes/wikia/Extensions.php [hunk 10]
```diff
@@ -0,0 +1,1892 @@
...
+
+/*
+ * Send email from the app authenticated by a secret token
+ */
+if( file_exists("$IP/extensions/wikia/SpecialEmailTest/SpecialEmailTest.php") ) {
+	include( "$IP/extensions/wikia/SpecialEmailTest/SpecialEmailTest.php" );
+}
+
+if ( !empty( $wgEnablePerSkinParserCacheExt ) && file_exists( "$IP/extensions/wikia/PerSkinParserCache/PerSkinParserCache.php" ) ) {
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:1145]: file existence check not needed

File: includes/wikia/Extensions.php [hunk 11]
```diff
@@ -0,0 +1,1892 @@
...
+$wgEnableOpenXSPC = true;
+$wgEnableSpotlightsV2_GlobalNav = true;
+$wgEnableSpotlightsV2_Footer = true;
+$wgEnableSpotlightsV2_Rail = true;
+
+/*
+ * Send email from the app authenticated by a secret token
+ */
+if( file_exists("$IP/extensions/wikia/SpecialEmailTest/SpecialEmailTest.php") ) {
+	include( "$IP/extensions/wikia/SpecialEmailTest/SpecialEmailTest.php" );
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:1142]: are we using this?

File: includes/wikia/Extensions.php [hunk 12]
```diff
@@ -0,0 +1,1892 @@
...
+	include( "$IP/extensions/wikia/OpenGraphMetaCustomizations/OpenGraphMetaCustomizations.setup.php");
+}
+
+if ( !empty( $wgEnableChat ) ){
+	include( "$IP/extensions/wikia/Chat2/Chat_setup.php" );
+}
+
+if ( !empty( $wgEnableAntiSpoofExt ) ) {
+	include( $IP . '/extensions/AntiSpoof/AntiSpoof.php' );
+//	$wgSharedTables[] = 'spoofuser';
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:1184]: let's remove commented code

File: includes/wikia/Extensions.php [hunk 13]
```diff
@@ -0,0 +1,1892 @@
...
+if ( !empty( $wgEnableAppPromoLandingExt ) ) {
+	include "$IP/extensions/wikia/AppPromoLanding/AppPromoLanding_setup.php";
+}
+
+/**
+ * @name $wgEnableNewAuthModal
+ * Enables new popup-based authentication
+ */
+if ( !isset($wgEnableNewAuthModal) && in_array( $wgLanguageCode, [ 'es', 'ru' ] ) ) {
+	$wgEnableNewAuthModal = true;
```
Comment (mszabo-wikia) [inline includes/wikia/Extensions.php:1860]: I believe this is now rolled out globally so this conditional can be removed

**Review by mszabo-wikia — DISMISSED** [review]

**Comment by ghost** [comment]
@TK-999 - the cleanup will be a separate part of the effort.  Right now I simply want to move existing stuff to proper places, without actually changing it too much.  I would like to merge self-contained things as the effort progresses, to avoid endless conflicts and one big PR with changes that will need to be updated before they can get tested thoroughly.

**Review by mszabo-wikia — APPROVED** [review]

**Pull request closed without merging**